spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Fusion.scala

```diff
@@ -98,7 +98,10 @@ private[mkldnn] object Fusion {
 
   private def findPrevious(node: Node[AbstractModule[Activity, Activity, Float]])
   : Node[AbstractModule[Activity, Activity, Float]] = {
-    if (node.element.isInstanceOf[Identity] && node.prevNodes.length == 1) {
+    if (node.element.isInstanceOf[Identity]
+      && node.nextNodes.length == 1
+      && node.prevNodes.length == 1
+      && node.prevNodes(0).nextNodes.length == 1) {
```

Comment:
Why should we add line 104 here ?

Reply:
This line is for cases like below (mentioned by @zhangxiaoli73 ):
```
                        conv1(96 -> 96)
                               |
                               |
                        conv2(96 -> 24)
                              /\        
                            /    \      
            conv3(24 ->144)    Identity    conv4(96 -> 24)
                                        \    /
                                          \/
                                        CAddTable
```

Reply:
I am not sure if there are any possible corner cases not taken into account yet.

Reply:
It's OK. But we should add at least two unit tests here for both cases.

Reply:
Added tests and added a condition for the case shown in the graph at the very beginning. I mean the convs without following relu, bn, etc.